x-pack/plugins/triggers_actions_ui/public/common/types.ts

```diff
@@ -5,17 +5,29 @@
  * 2.0.
  */
 
+import { KBN_FIELD_TYPES } from '@kbn/field-types';
+
 export interface Comparator {
   text: string;
   value: string;
   requiredValues: number;
 }
 
+export type ValidNormalizedTypes = `${Exclude<
```

Comment:
The main problem with this is that if new `KBN_FIELD_TYPES` get added, they will be considered "valid".  Feels like we should explicitly add the types we want to support, vs removing the ones we don't want (right now).

Reply:
Do we have a list of the valid types? I wanted to add some additional types to the `cardinality` aggregation but I wasn't sure which types were valid, I was hoping that would have been defined in Typescript. When I saw that it wasn't, I went down the rabbit hole of finding `KBN_FIELD_TYPES` and then working to set `validNormalizedTypes` to a union of the values of the enumeration.

Reply:
I don't know, but guessing probably no.  No need for this to hold up the PR, but could you open an issue to track, assuming you don't find anything else in the next few minutes?